packages/loopring_v3/contracts/amm/libamm/AmmStatus.sol

```diff
@@ -71,16 +71,26 @@ library AmmStatus
             }));
         }
 
-        // Mint all liquidity tokens to the pool account on L2
-        S.balanceOf[address(this)] = AmmData.POOL_TOKEN_MINTED_SUPPLY;
-        S.allowance[address(this)][address(exchange.getDepositContract())] = type(uint256).max;
-        exchange.deposit(
-            address(this), // from
-            address(this), // to
-            address(this), // token
-            uint96(AmmData.POOL_TOKEN_MINTED_SUPPLY),
-            new bytes(0)
-        );
+        bool newPool = (S.accountID == 0);
+        if (newPool) {
```

Comment:
This logic seems incorrect to me. No matter if this is a new pool or an existing pool to be reused, we just need to check that the LP token balance is 0. For reusable pools, we may need to force the pool to use the same accountID, as the pool's address is the same, so its accountID should stay unchanged.


```solidity
require(S.poolTokenBalance == 0),
if (S.accountID == 0) { // new pool
  require (config.accountID > 0);
  S.accountID = config.accountID;
} else { // reused pool
  require(S.shutdownTimestamp > 0);
  require(config.accountID == 0 || config.accountID == S.accountID)
  S.shutdownTimestamp = 0;
  S.exitMode = false;
}
// all other parameter setup
```

Reply:
> This logic seems incorrect to me. No matter if this is a new pool or an existing pool to be reused, we just need to check that the LP token balance is 0.

I think that's what my code also does?

> For reusable pools, we may need to force the pool to use the same accountID, as the pool's address is the same, so its accountID should stay unchanged.

The current code only sets the accountID for new pools, it ignores the passed in value in `config` when the pool is reused.

I don't think a requirement for pool reuse is that the pool _needs_ to have been shutdown. But otherwise your code and the current code seems the same or did I miss anything?

Reply:
I'll propose a change with a question in another PR